src/ManagedServices/ManagedServices/Commands/RemoveAzureRmManagedServicesDefinition.cs

```diff
@@ -0,0 +1,73 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using Microsoft.Azure.PowerShell.Cmdlets.ManagedServices.Models;
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+using System.Text;
+
+namespace Microsoft.Azure.PowerShell.Cmdlets.ManagedServices.Commands
+{
+    [Cmdlet(
+    VerbsCommon.Remove,
+    Microsoft.Azure.Commands.ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "ManagedServicesDefinition",
+    DefaultParameterSetName = DefaultParameterSet,
+    SupportsShouldProcess = true), OutputType(typeof(PSRegistrationDefinition))]
+
+    public class RemoveAzureRmManagedServicesDefinition : ManagedServicesCmdletBase
+    {
+        protected const string InputObjectParameterSet = "InputObject";
+        protected const string DefaultParameterSet = "Default";
+
+        [Parameter(Position = 0, ParameterSetName = DefaultParameterSet, Mandatory = false, HelpMessage = "The subscription scope.")]
+        [ScopeCompleter]
+        public string Scope { get; set; }
+
+        [Parameter(Position = 1, ParameterSetName = DefaultParameterSet, Mandatory = true, HelpMessage = "The registration definition identifier.")]
+        [Alias("Id")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
```

Comment:
You need to have two parameter sets:
- one which breaks fown the resource id by its components (for most resources, this is resource group _ name, in this case, it may just be name, if these are tenant or subscription-level resources.

- one which takes ResourceId from a call to Get-AzResource with value from pipelinebypropertyname